### PR TITLE
TST: Add to_csv test when writing the single column CSV

### LINKED ISSUE #18676: to_csv writes wrong with NaN value
#### to_csv with Nan value at top row, unexpected "" in the csv file

```python
import pandas as pd
df = pd.DataFrame([None,1,2])
df.to_csv("df.csv",header=None,index=None,encoding ='utf-8')

# df.csv
#""
#1.0
#2.0

# I want to make df.csv
#
#1.0
#2.0
```

#### Versions:
```
pandas.show_versions()

INSTALLED VERSIONS

commit: None
python: 3.6.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.3
pytest: None
pip: 9.0.1
setuptools: 36.4.0
Cython: None
numpy: 1.13.1
scipy: 0.19.1
xarray: None
IPython: 6.1.0
sphinx: None
patsy: None
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None
​
```

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #18676
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/test_to_csv.py [hunk 1]
```diff
@@ -9,6 +10,34 @@
 
 class TestToCSV(object):
 
+    @pytest.mark.xfail((3, 6, 5) > sys.version_info >= (3, 5),
+                       reason='Python csv library bug')
+    def test_to_csv_with_single_column(self):
+        # GH18676
+        df1 = DataFrame([None, 1],
+                        header=None,
```
Comment (gfyoung) [inline pandas/tests/io/formats/test_to_csv.py]: Oops.  I think you've been in CSV-land for too long? 😄

**Comment by gfyoung** [comment]
@Licht-T : Overall, well done on patching this bug in its entirety across multiple repositories!

**Comment by Licht-T** [comment]
Re-commit because of the CI falure

File: pandas/tests/io/formats/test_to_csv.py [hunk 2]
```diff
@@ -9,6 +10,30 @@
 
 class TestToCSV(object):
 
+    @pytest.mark.xfail((3, 6, 5) > sys.version_info >= (3, 5),
+                       reason='Python csv library bug')
```
Comment (jreback) [inline pandas/tests/io/formats/test_to_csv.py]: can you put a refernce to the bug, and a comment on what it is

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment

**Comment by jreback** [comment]
can you rebase / update

**Comment by gfyoung** [comment]
@jreback : I can carry this to the finish line if need be.

File: pandas/tests/io/formats/test_to_csv.py (hunk 2 above)
Comment (gfyoung) [reply pandas/tests/io/formats/test_to_csv.py]: Done.

Comment (jreback) [reply pandas/tests/io/formats/test_to_csv.py]: can you also add the link to the xfail reason

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gfyoung tiny comment. pls fix and merge away

**Comment by gfyoung** [comment]
@jreback : Done.  Will wait until green before merging.

**Comment by gfyoung** [comment]
Alright, all is green.  Merging this.

**Comment by gfyoung** [comment]
Thanks @Licht-T for your hard work!

**Pull request merged**